contracts/ecash/src/contract.rs

```diff
@@ -0,0 +1,430 @@
+    SubMsg, WasmMsg,
+};
+use cw3::ProposalResponse;
+use cw4::Cw4Contract;
+use cw_controllers::Admin;
+use cw_storage_plus::{Bound, IndexedMap, Item};
+use nym_ecash_contract_common::blacklist::{
+    BlacklistProposal, BlacklistedAccount, BlacklistedAccountResponse,
+    PagedBlacklistedAccountResponse,
+};
+use nym_ecash_contract_common::events::{
+    BLACKLIST_PROPOSAL_ID, BLACKLIST_PROPOSAL_REPLY_ID, DEPOSITED_FUNDS_EVENT_TYPE,
+    DEPOSIT_ENCRYPTION_KEY, DEPOSIT_IDENTITY_KEY, DEPOSIT_INFO, DEPOSIT_VALUE,
+};
+use nym_ecash_contract_common::msg::ExecuteMsg;
+use nym_multisig_contract_common::msg::ExecuteMsg as MultisigExecuteMsg;
+use nym_multisig_contract_common::msg::QueryMsg as MultisigQueryMsg;
+use sylvia::types::{ExecCtx, InstantiateCtx, QueryCtx, ReplyCtx};
+use sylvia::{contract, entry_points};
+
+use crate::errors::ContractError;
+use crate::state::Config;
+use crate::storage::{
+    self, BlacklistIndex, BlacklistProposalIndex, SpendCredentialIndex,
+    BLACKLIST_PAGE_DEFAULT_LIMIT, BLACKLIST_PAGE_MAX_LIMIT, SPEND_CREDENTIAL_PAGE_DEFAULT_LIMIT,
+    SPEND_CREDENTIAL_PAGE_MAX_LIMIT,
+};
+use nym_ecash_contract_common::events::{TICKET_BOOK_VALUE, TICKET_VALUE};
+use nym_ecash_contract_common::spend_credential::{
+    EcashSpentCredential, EcashSpentCredentialResponse, PagedEcashSpentCredentialResponse,
+};
+
+pub struct NymEcashContract<'a> {
+    pub(crate) admin: Admin<'a>,
+    pub(crate) config: Item<'a, Config>,
+    pub(crate) spent_credentials:
+        IndexedMap<'a, &'a str, EcashSpentCredential, SpendCredentialIndex<'a>>,
+    pub(crate) blacklist: IndexedMap<'a, &'a str, BlacklistedAccount, BlacklistIndex<'a>>,
+    pub(crate) blacklist_proposals:
+        IndexedMap<'a, &'a str, BlacklistProposal, BlacklistProposalIndex<'a>>,
+}
+
+#[entry_points]
+#[contract]
+#[error(ContractError)]
+impl NymEcashContract<'_> {
+    pub const fn new() -> Self {
+        Self {
+            admin: Admin::new("admin"),
+            config: Item::new("config"),
+            spent_credentials: storage::spent_credentials(),
+            blacklist: storage::blacklist(),
+            blacklist_proposals: storage::blacklist_proposal(),
+        }
+    }
+
+    #[msg(instantiate)]
+    pub fn instantiate(
+        &self,
+        mut ctx: InstantiateCtx,
+        multisig_addr: String,
+        group_addr: String,
+        mix_denom: String,
+    ) -> Result<Response, ContractError> {
+        let multisig_addr = ctx.deps.api.addr_validate(&multisig_addr)?;
+        let group_addr = Cw4Contract(ctx.deps.api.addr_validate(&group_addr).map_err(|_| {
+            ContractError::InvalidGroup {
+                addr: group_addr.clone(),
+            }
+        })?);
+
+        self.admin
+            .set(ctx.deps.branch(), Some(multisig_addr.clone()))?;
+        let cfg = Config {
+            multisig_addr,
+            group_addr,
+            mix_denom,
+        };
+
+        self.config.save(ctx.deps.storage, &cfg)?;
+
```

Comment:
could you also set the cw2 version?
Oh and ideally our full build info we have in mixnet and vesting contracts

Reply:
Added version in 34c5c6a6201faa79e00e9b68fe18aef738e65d26